app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,37 @@
+class LikesController < ApplicationController
+  before_action :find_event
+  before_action :find_like, only: [:destroy]
+
+  def create
+    if already_liked?
+      flash[:notice] = "You can't like more than once"
+    else
+      @event.likes.create(user_id: current_user.id)
+    end
+    redirect_to event_path(@event)
+  end
+
+  def destroy
+    if !(already_liked?)
+      flash[:notice] = "Cannot unlike"
+    else
+      @like.destroy
+    end
+    redirect_to event_path(@event)
+  end
+
+  private
+
+  def find_event
+    @event = Event.find(params[:event_id])
+  end
```

Comment:
Why add this in likes controller?

Reply:
because to be able to like, you must first see the event you are going to like, and that's why the find_event method.

Reply:
I suggest you use a decorator, so instead of naming this `find_event` you name it `event` and use `@event` to memoize

Reference:  
https://nithinbekal.com/posts/ruby-decorators/

Reply:
ok, i'll see this

Reply:
wouldn't the before_action be better with the find_event

Reply:
OK, lets continue with find_event